### PR TITLE
PLANNER-2216 Add Optawebs to the release pipeline

### PR DESCRIPTION
**Pull request by rsynek**
Tested with https://rhba-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/custom/job/rsynek/job/releasePipeline/job/kogito-release/28/.

### JIRA
https://issues.redhat.com/browse/PLANNER-2216

### Referenced pull requests
https://github.com/kiegroup/kogito-pipelines/pull/90

### Checklist
- [ ] Documentation updated if applicable.
- [ ] Upgrade recipe provided if applicable.

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* for a <b>pull request build</b> please add comment: <b>Jenkins retest this</b>
* for a <b>full downstream build</b> please add comment: <b>Jenkins run fdb</b>
* for a <b>compile downstream build</b> please add comment: <b>Jenkins run cdb</b>
* for a <b>full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>
* for an <b>upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

File: Jenkinsfile.promote [hunk 1]
```diff
@@ -77,34 +81,126 @@ pipeline {
...
-                    dir('optaplanner') {
+                    dir(optaplannerRepository) {
+                        mavenCleanInstall(true, [], '-Dfull')
                         withCredentials(bindings: [sshUserPrivateKey(credentialsId: 'optaplanner-filemgmt',
                                              keyFileVariable: 'SSH_KEY_JBOSS_FILEMGMT')]) {
                             // For testing, simulate connection via SSH:
-                            // sh "ssh -i $SSH_KEY_JBOSS_FILEMGMT -oKexAlgorithms=+diffie-hellman-group1-sha1 optaplanner@filemgmt.jboss.org"
-                            sh "./build/release/upload_distribution.sh ${getProjectVersion()} $SSH_KEY_JBOSS_FILEMGMT"
+                            sh "ssh -i $SSH_KEY_JBOSS_FILEMGMT -oKexAlgorithms=+diffie-hellman-group1-sha1 optaplanner@filemgmt.jboss.org"
+                            // TODO: return back
```
Comment (rsynek) [inline Jenkinsfile.promote:162]: TODO: return back after testing.

File: Jenkinsfile.promote [hunk 2]
```diff
@@ -77,34 +81,126 @@ pipeline {
...
+            }
+            steps {
+                script {
+                    dir(vehicleRoutingRepository) {
+                        mavenCleanInstall(true)
+                        withCredentials(bindings: [sshUserPrivateKey(credentialsId: 'optaplanner-filemgmt',
+                                keyFileVariable: 'SSH_KEY_JBOSS_FILEMGMT')]) {
+                            // For testing, simulate connection via SSH:
+                            sh "ssh -i $SSH_KEY_JBOSS_FILEMGMT -oKexAlgorithms=+diffie-hellman-group1-sha1 optaplanner@filemgmt.jboss.org"
+                            // TODO: return back
```
Comment (rsynek) [inline Jenkinsfile.promote:182]: TODO: return back after testing.

File: Jenkinsfile.promote [hunk 3]
```diff
@@ -77,34 +81,126 @@ pipeline {
...
+            }
+            steps {
+                script {
+                    dir(employeeRosteringRepository) {
+                        mavenCleanInstall(true)
+                        withCredentials(bindings: [sshUserPrivateKey(credentialsId: 'optaplanner-filemgmt',
+                                keyFileVariable: 'SSH_KEY_JBOSS_FILEMGMT')]) {
+                            // For testing, simulate connection via SSH:
+                            sh "ssh -i $SSH_KEY_JBOSS_FILEMGMT -oKexAlgorithms=+diffie-hellman-group1-sha1 optaplanner@filemgmt.jboss.org"
+                            // TODO: return back
```
Comment (rsynek) [inline Jenkinsfile.promote:202]: TODO: return back after testing.

File: Jenkinsfile.deploy [hunk 4]
```diff
@@ -263,6 +277,13 @@ void commitAndCreatePR(String repo) {
     }
 }
 
+void commitAndCreatePRIgnoringPackageLock(String repo) {
+    dir(repo) {
+        sh 'git checkout -- */package-lock.json'
```
Comment (yurloc) [inline Jenkinsfile.deploy:282]: Changes in `package-lock.json` should not be ignored.

Comment (rsynek) [reply Jenkinsfile.deploy:282]: Good point. We should commit the changes as these are exact versions we built with, right?

File: Jenkinsfile.promote [hunk 5]
```diff
@@ -77,34 +81,126 @@ pipeline {
...
+                    dir(optaplannerRepository) {
+                        mavenCleanInstall(true, [], '-Dfull')
                         withCredentials(bindings: [sshUserPrivateKey(credentialsId: 'optaplanner-filemgmt',
                                              keyFileVariable: 'SSH_KEY_JBOSS_FILEMGMT')]) {
                             // For testing, simulate connection via SSH:
-                            // sh "ssh -i $SSH_KEY_JBOSS_FILEMGMT -oKexAlgorithms=+diffie-hellman-group1-sha1 optaplanner@filemgmt.jboss.org"
-                            sh "./build/release/upload_distribution.sh ${getProjectVersion()} $SSH_KEY_JBOSS_FILEMGMT"
+                            sh "ssh -i $SSH_KEY_JBOSS_FILEMGMT -oKexAlgorithms=+diffie-hellman-group1-sha1 optaplanner@filemgmt.jboss.org"
+                            // TODO: return back
+                            //sh "./build/release/upload_distribution.sh ${getProjectVersion()} $SSH_KEY_JBOSS_FILEMGMT"
```
Comment (radtriste) [inline Jenkinsfile.promote:163]: I say this (ssh + script call) could be factorized as you are executing 3 times the same or ?
even better as you are also using the creds part

**Review by radtriste — APPROVED** [review]
looks good. one possible improvement.

**Comment by rsynek** [comment]
Running hopefully the last test of the pipeline.

**Comment by rsynek** [comment]
The test failure cannot be related to this PR. Maybe unstable tests?

**Comment by rsynek** [comment]
jenkins retest this

**Comment by rsynek** [comment]
Cannot reproduce locally

**Comment by rsynek** [comment]
jenkins retest this

**Pull request merged**